velox/benchmarks/basic/LikeFunctionsBenchmark.cpp

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+
+#include "vector/fuzzer/VectorFuzzer.h"
+#include "velox/functions/lib/Re2Functions.h"
+#include "velox/functions/lib/benchmarks/FunctionBenchmarkBase.h"
+#include "velox/functions/prestosql/tests/FunctionBaseTest.h"
+#include "velox/tpch/gen/TpchGen.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::tpch;
+using namespace facebook::velox::functions;
+using namespace facebook::velox::functions::test;
+
+DEFINE_int32(vector_size, 10000, "Vector size");
+DEFINE_int32(num_rows, 10000, "Number of rows");
+
+namespace {
+
+class LikeFunctionsBenchmark : public FunctionBaseTest,
+                               public FunctionBenchmarkBase {
+ public:
+  explicit LikeFunctionsBenchmark() {
+    exec::registerStatefulVectorFunction("like", likeSignatures(), makeLike);
+
+    folly::BenchmarkSuspender kSuspender;
```

Comment:
The suspender is used to avoid counting some work in the benchmarking time (like stop the timer and re-enable it).
since this part is not running in a  benchmark, but pre-proccessign its not needed here. processing

Reply:
Thanks for the clarification, removed it from here and used it in the scope of input data generation for benchmarking